lib/galaxy/datatypes/sniff.py

```diff
@@ -597,6 +598,21 @@ def search(self, pattern):
     def search_str(self, query_str):
         return query_str in self.contents_header
 
+    def unpack_header(self, pattern):
+        size = struct.calcsize(pattern)
+        header_bytes = self.contents_header_bytes[0:size]
+        if len(header_bytes) < size:
+            return None
+        return struct.unpack(pattern, header_bytes)
+
+    def magic_header(self, pattern):
+        # unpack header and get first element
+        unpacked = self.unpack_header(pattern)
+        return None if not unpacked else unpacked[0]
```

Comment:
```suggestion
    def magic_header(self, pattern):
    	"""
        Unpack header and get first element
        """
        size = struct.calcsize(pattern)
        header_bytes = self.contents_header_bytes[:size]
        if len(header_bytes) < size:
            return None
        return struct.unpack(pattern, header_bytes)[0]
```

Reply:
Comment indent looks incorrect in github preview.

Reply:
Thanks @martenson , fixed.